<launch>
    <arg name="marker_size" default="5.3" />
    <arg name="max_new_marker_error" default="0.08" />
    <arg name="max_track_error" default="0.2" />

    <arg name="cam_image_topic" default="/usb_cam/image_raw" />
    <arg name="cam_info_topic" default="/usb_cam/camera_info" />
    <arg name="output_frame" default="/base" />

    <!-- usb_cam arguments -->
    <arg name="video_device" default="/dev/video0" />
    <arg name="pixel_format" default="yuyv" />
    <arg name="camera_name"  default="tracker_camera" />

    <node name="ar_track_alvar" pkg="ar_track_alvar" type="individualMarkersNoKinect" respawn="false" output="screen">
        <param name="marker_size"           type="double" value="$(arg marker_size)" />
        <param name="max_new_marker_error"  type="double" value="$(arg max_new_marker_error)" />
        <param name="max_track_error"       type="double" value="$(arg max_track_error)" />
        <param name="output_frame"          type="string" value="$(arg output_frame)" />

        <remap from="camera_image"  to="$(arg cam_image_topic)" />
        <remap from="camera_info"   to="$(arg cam_info_topic)" />
    </node>

  <!-- USB Camera Driver -->
  <node name="usb_cam" pkg="usb_cam" type="usb_cam_node" />
    <param name="usb_cam/video_device" value="$(arg video_device)" />
    <param name="usb_cam/pixel_format" value="$(arg pixel_format)" />
    <param name="usb_cam/camera_name" value="$(arg camera_name)" />

    <node name="rviz" pkg="rviz" type="rviz" />


    <node name="base_to_head_camera" pkg="tf" type="static_transform_publisher" args="0 0 0 0 0 0 base head_camera 100" />




</launch>
